tests/framework/results/test_context.py

```diff
@@ -443,3 +445,130 @@ def test__bad_aggregator_stratification():
     with pytest.raises(KeyError, match="height"):
         for result, _measure in ctx.gather_results(population, event_name):
             print(result)
+
+
+@pytest.mark.parametrize("pop_filter", ['familiar=="spaghetti_yeti"', ""])
+def test__filter_population(pop_filter):
+    filtered_pop = ResultsContext()._filter_population(
+        population=BASE_POPULATION, pop_filter=pop_filter
+    )
+    if pop_filter:
+        # The pop filter is set up to filter out everyone
+        assert filtered_pop.empty
+    else:
+        # An empty pop filter should return the entire population
+        assert filtered_pop.equals(BASE_POPULATION)
+
+
+@pytest.mark.parametrize("stratifications", [("familiar",), ()])
```

Comment:
Would it be a better test if there were two stratification rather than just one? (e.g. familiar and house)